pandas/core/window/rolling.py

```diff
@@ -1959,7 +1959,10 @@ def validate(self):
 
             # this will raise ValueError on non-fixed freqs
             self.win_freq = self.window
-            self.window = freq.nanos
+            if isinstance(self._on, ABCPeriodIndex):
```

Comment:
Can you create a helper method to do this instead?

Reply:
Thx, something like this?